pkg/controller/controller.go

```diff
@@ -61,6 +64,12 @@ func (c *AppGwIngressController) Start(envVariables environment.EnvVariables) er
 		return err
 	}
 
+	if pod, err := c.k8sContext.GetAGICPod(envVariables); err == nil {
```

Comment:
This entire block should be combined with the implementation of the `GetAGICPod` and moved as a one-liner somewhere above this: https://github.com/Azure/application-gateway-kubernetes-ingress/blob/master/cmd/appgw-ingress/main.go#L128

This should happen once only during the lifetime of the pod, and it makes sense to put it near the initialization of the Controller.  We could pass the Pod identity into `NewAppGwIngressController`

Reply:
sure